Added rules for pytest fixtures:

DKorytkin opened the pull request:
- rule for deprecated fixture scope as a positional argument ([link](https://docs.pytest.org/en/stable/reference.html#pytest-fixture))
- rule for useless `pytest.mark.usefixtures` for fixtures ([link](https://docs.pytest.org/en/stable/fixture.html#usefixtures))
  <img width="875" alt="Screenshot 2021-03-24 at 17 56 26" src="https://user-images.githubusercontent.com/12531847/112341634-42630480-8cca-11eb-8ad1-7b8749ffe66b.png">

- added tests for these checks
- removed `py27` and `py35` from testing, seems not supported

DKorytkin committed:
Added rules to pytest fixtures:
- rule for deprecated fixture scope as positional argument
- rule for useless pytest.mark.usefixtures for fixtures
- added tests for these checks

DKorytkin committed:
added to readme

reverbc commented on pylint_pytest/checkers/fixture.py:
```diff
@@ -91,6 +104,42 @@ def visit_module(self, node):
+        ----------
+        node : astroid.scoped_nodes.Decorators
+        """
+        uses_fixture_deco, uses_usefixture_deco = 0, 0
```
I'd prefer we don't use `int` for these two variables. We're not using the count anywhere in the below code and they can be replaced with `bool` and
```python
uses_fixture_deco |= _is_pytest_fixture(decorator)
uses_usefixture_deco |= _is_pytest_mark_usefixtures(decorator)
```

reverbc commented on pylint_pytest/checkers/fixture.py:
```diff
@@ -25,6 +25,19 @@ class FixtureChecker(BasePytestChecker):
+            'useless-pytest-usefixtures-decorator',
+            (
+                '@pytest.mark.usefixtures decorator doesn\'t work as expected for fixtures. '
+                'Take a look at: https://docs.pytest.org/en/stable/fixture.html#usefixtures'
```
According to https://docs.pytest.org/en/stable/reference.html#marks, all `@pytest.mark` are not supported on fixtures.

reverbc commented on tox.ini:
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py27,py35,py36,py37,py38
+envlist = py36,py37,py38
```
✨🍰✨

Would you mind to also include `py39`?

reverbc commented on pylint_pytest/checkers/fixture.py:
```diff
@@ -25,6 +25,19 @@ class FixtureChecker(BasePytestChecker):
+        'W6403': (
+            'Using a deprecated scope of fixture as positional argument',
+            'deprecated-pytest-fixture-scope-as-positional-argument',
+            'Pass scope as a kwarg, not positional arg, which is deprecated in future pytest.',
```
All positional arguments, not just `scope`, are prohibited since pytest 6.0.0 https://docs.pytest.org/en/stable/deprecations.html#pytest-fixture-arguments-are-keyword-only

reverbc reviewed (changes requested):
Thanks for the work @DKorytkin! Generally it looks fine and I believe they are great additional functionalities to the plugin, and I think we can do more with both warnings:

For `W6402 useless-pytest-usefixtures-decorator`, I took a deeper look at the `@pytest.mark.usefixtures` issue, and it turns out `@pytest.mark` does not work on fixtures at all:

>Marks
> Marks can be used apply meta data to test functions (but not fixtures), which can then be accessed by fixtures or plugins.
- https://docs.pytest.org/en/stable/reference.html#marks

For `W6403 deprecated-pytest-fixture-scope-as-positional-argument`, all positional arguments to `@pytest.fixture` are prohibited since Pytest 6.0.0: 

> pytest.fixture arguments are keyword only
> _Removed in version 6.0._
>
> Passing arguments to pytest.fixture() as positional arguments has been removed - pass them by keyword instead.
- https://docs.pytest.org/en/stable/deprecations.html#pytest-fixture-arguments-are-keyword-only

Can you increase the scope to those two warnings so they can be even more helpful?

DKorytkin committed:
Fixed comment in PR:
- used check for all `@pytest.mark.*` decorators
- edited error message for warnning W6403
- added py39 for testing

DKorytkin commented:
@reverbc thanks for the review, I fixed all your suggestions 👍 
can you take a look one more time 😉

reverbc reviewed (approved):
Thanks for the contribution!
